src/Features/CSharp/Portable/CodeFixes/Nullable/DeclareAsNullableCodeFixProvider.cs

```diff
@@ -0,0 +1,135 @@
+        {
+            // warning CS8625: Cannot convert null literal to non-nullable reference or unconstrained type parameter.
+            // warning CS8600: Converting null literal or possible null value to non-nullable type.
+            get { return ImmutableArray.Create("CS8625", "CS8600"); }
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.First();
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+
+            var declarationTypeToFix = TryGetDeclarationTypeToFix(node);
+            if (declarationTypeToFix == null)
+            {
+                return;
+            }
+
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(context.Document, diagnostic, c)),
+                context.Diagnostics);
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var root = editor.OriginalRoot;
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+                MakeDeclarationNullable(document, editor, node);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        internal static void MakeDeclarationNullable(Document document, SyntaxEditor editor, SyntaxNode node)
+        {
+            var declarationTypeToFix = TryGetDeclarationTypeToFix(node);
+            if (declarationTypeToFix != null)
+            {
+                var fixedDeclaration = SyntaxFactory.NullableType(declarationTypeToFix);
+                editor.ReplaceNode(declarationTypeToFix, fixedDeclaration);
+            }
+        }
+
+        private static TypeSyntax TryGetDeclarationTypeToFix(SyntaxNode node)
+        {
+            if (!node.IsKind(SyntaxKind.NullLiteralExpression))
+            {
+                return null;
+            }
+
+            // return null;
+            if (node.IsParentKind(SyntaxKind.ReturnStatement))
+            {
+                var methodDeclaration = node.GetAncestors<MethodDeclarationSyntax>().FirstOrDefault();
+                if (methodDeclaration != null)
+                {
+                    return methodDeclaration.ReturnType;
+                }
+            }
+
+            // string x = null;
+            if (node.Parent?.Parent?.IsParentKind(SyntaxKind.VariableDeclaration) == true)
+            {
+                var variableDeclaration = (VariableDeclarationSyntax)node.Parent.Parent.Parent;
+                if (variableDeclaration.Variables.Count != 1)
```

Comment:
note: there are probably other cases that need to be supported, such as a for each statement. if you don't implement this, please create an issue so that somebody can fix this later

Reply:
I can't think of any other scenario at the moment. But we can always fill any gaps we discover. I'll be testing this on Roslyn codebase, so that may give some ideas.